### PR TITLE
replace ramda with welshman/lib

### PR DESCRIPTION
**Pull request by ticruz38**
#372 

Replaced most of ramda utilities with either welshman/lib or standard JS functions.

takeWhile might deserve its own implementation in welshman/lib.

### CODE REVIEW

File: src/app/App.svelte [hunk 1]
```diff
@@ -353,7 +352,7 @@
     } else if (document.body.style.position === "fixed") {
       document.body.setAttribute("style", "")
 
-      if (!isNil(scrollY)) {
+      if (scrollY) {
```
Comment (staab) [inline src/app/App.svelte:355]: This is important in the case of scrollY being 0; welshman/lib has an isNil function

File: src/app/Routes.svelte [hunk 2]
```diff
@@ -74,7 +73,7 @@
   </div>
 {/key}
 
-{#each reverse($modals).filter(m => !m.virtual) as m, i (router.getKey(m) + i)}
+{#each $modals.reverse().filter(m => !m.virtual) as m, i (router.getKey(m) + i)}
```
Comment (staab) [inline src/app/Routes.svelte:76]: This mutates the variable, which is definitely not what we want.

File: src/app/shared/Channel.svelte [hunk 3]
```diff
@@ -86,12 +85,12 @@
   const scrollToBottom = () => element.scrollIntoView({behavior: "smooth", block: "end"})
 
   const stickToBottom = async () => {
-    const lastMessage = pluck("created_at", groupedMessages).reduce(max, 0)
+    const lastMessage = max(groupedMessages.map(prop("created_at")) as number[])
```
Comment (staab) [inline src/app/shared/Channel.svelte:88]: Prefer lambdas over type assertions. Prop is probably a bad idea.

File: src/app/views/Zap.svelte [hunk 4]
```diff
@@ -17,60 +15,53 @@
   export let splits
   export let id = null
   export let anonymous = false
-  export let amount = getSetting("default_zap")
+  export let amount = getSetting("default_zap") as string
```
Comment (staab) [inline src/app/views/Zap.svelte:18]: Assertion is incorrect, it should be a number right?

File: src/engine/state.ts [hunk 5]
```diff
@@ -562,7 +566,7 @@ export const withIndexers = (relays: string[]) => withRelays(relays, env.INDEXER
 export const lists = deriveEventsMapped<PublishedUserList>(repository, {
   filters: [{kinds: EDITABLE_LIST_KINDS}],
   eventToItem: (event: TrustedEvent) => (event.tags.length > 1 ? readUserList(event) : null),
-  itemToEvent: prop("event"),
+  itemToEvent: prop("event") as any,
```
Comment (staab) [inline src/engine/state.ts:569]: Use a lambda here instead of `any`

File: src/engine/state.ts [hunk 6]
```diff
@@ -726,7 +730,7 @@ export const collectionSearch = derived(
 // Network
 
 export const getExecutor = (urls: string[]) => {
-  const [localUrls, remoteUrls] = partition(equals(LOCAL_RELAY_URL), urls)
+  const [localUrls, remoteUrls] = partition(url => equals(LOCAL_RELAY_URL, url), urls)
```
Comment (staab) [inline src/engine/state.ts:733]: `===` is fine here, equals doesn't add anything for strings

File: src/util/router.ts [hunk 7]
```diff
@@ -208,9 +208,9 @@ class RouterExtension {
     return this.clone({queryParams: data})
   }
 
-  cx = context => this.clone(updateIn("context", mergeLeft(context))(this.params))
+  cx = context => this.clone(updateIn("context", (c: object) => ({...c, ...context}))(this.params))
 
-  cg = config => this.clone(updateIn("config", mergeLeft(config))(this.params))
+  cg = config => this.clone(updateIn("config", (c: object) => ({...c, ...config}))(this.params))
```
Comment (staab) [inline src/util/router.ts:213]: mergeLeft is available in welshman

**Review by staab — COMMENTED** [review]
- I've released a new version of welshman with `pluck` in it, go ahead and use that instead of map/prop. Also `spec` and `complement` can be useful to replace `whereEq`, but that's optional
- Prop is now parameterizable, so you can avoid type assertions
- I've added a linting rule for triple equals, be sure to rebase and lint

File: src/app/views/Zap.svelte (hunk 4 above)
Comment (ticruz38) [reply src/app/views/Zap.svelte:18]: yes

File: src/app/App.svelte [hunk 8]
```diff
@@ -435,7 +432,7 @@
         ])
 
         if (lastActivity && lastActivity < ago(30)) {
-          ctx.net.pool.get(connection.url).socket.close()
+          ctx.net.pool.remove(connection.url)
```
Comment (staab) [inline src/app/App.svelte:435]: Nope, don't revert this

**Review by staab — COMMENTED** [review]
One final thing and it's good to go

**Comment by staab** [comment]
Merged manually

**Pull request closed without merging**